core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -323,6 +326,27 @@ public AbstractClientOptionsBuilder setHeaders(
         return this;
     }
 
+    /**
+     * (Advanced user only) Sets the authoritative name of the remote server. This value is passed as the
+     * value of the {@code :authority} pseudo-header in HTTP/2 or the {@code "Host"} header in HTTP/1.1.
+     * If {@code null}, the authoritative name is automatically populated from one of the following values.
+     * <ul>
+     *   <li>A base {@link URI} specified when creating a {@link Client}.</li>
+     *   <li>An absolute path specified when sending a {@link Request}.</li>
+     *   <li>An {@link HttpHeaderNames#AUTHORITY} header specified in {@link RequestHeaders}.</li>
```

Comment:
Is it possible to also update the javadocs at `RequestHeadersBuilder` to reflect that
1) If a baseUri is defined the authority header is overwritten
2) If a baseUri is not defined, the authority header value is potentially used as the endpoint.